diff --git a/PKGBUILD b/PKGBUILD
index 76e4831..a7f217c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -69,7 +69,8 @@ source=("https://www.kernel.org/pub/linux/kernel/v5.x/linux-${_basekernel}.tar.x
         '0013-bootsplash.patch'
         'zfs-fix.patch'
         'fsync.patch'
-        'btrfs-prefer-nonrotational.patch')
+        'btrfs-prefer-nonrotational.patch'
+        'preempt.config')
 sha256sums=('a6fbd4ee903c128367892c2393ee0d9657b6ed3ea90016d4dc6f1f6da20b2330'
             '5aabcb81dcea06dca5c9daf739a0309eaf285365150f27419e0c00c5874c8511'
             '62cffd4cc10eb86de05c1a9a332c98289789ec18294a6331a6e42c9633a6411c'
@@ -116,7 +117,8 @@ sha256sums=('a6fbd4ee903c128367892c2393ee0d9657b6ed3ea90016d4dc6f1f6da20b2330'
             '035ea4b2a7621054f4560471f45336b981538a40172d8f17285910d4e0e0b3ef'
             '5d55bb51c3cade17dde2fec87a385604780afd117dbd01dfbb4b1b83f1ae2660'
             '2d9260b80b43bbd605cf420d6bd53aa7262103dfd77196ba590ece5600b6dc0d'
-            '2a47402a0fd9a0f9e6c9e83f25370e781b09c390d1d724320b756d7880c80416')
+            '2a47402a0fd9a0f9e6c9e83f25370e781b09c390d1d724320b756d7880c80416'
+            'a98c04e6a215a99bbb6bce3f572245235edcb0497d46fb33beea102fa4f25e36')
 prepare() {
   cd "${srcdir}/linux-${_basekernel}"
 
@@ -203,6 +205,9 @@ prepare() {
   # set extraversion to pkgrel
   sed -ri "s|^(EXTRAVERSION =).*|\1 -${pkgrel}|" Makefile
 
+  # apply custom config fragments
+  KCONFIG_CONFIG="./.config" "[merge_script]" -m "./.config" "${srcdir}/preempt.config"
+
   # don't run depmod on 'make install'. We'll do this ourselves in packaging
   sed -i '2iexit 0' scripts/depmod.sh
 
