diff --git a/PKGBUILD b/PKGBUILD
index 8674429..8cfc279 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -55,7 +55,8 @@ source=("https://www.kernel.org/pub/linux/kernel/v5.x/linux-${_basekernel}.tar.x
         '0013-bootsplash.patch'
         'zfs-fix.patch'
         'fsync.patch'
-        'btrfs-prefer-nonrotational.patch')
+        'btrfs-prefer-nonrotational.patch'
+        'preempt.config')
 sha256sums=('e342b04a2aa63808ea0ef1baab28fc520bd031ef8cf93d9ee4a31d4058fcb622'
             'a677aa98df0fc907314130b511feef3108717d98c405c98a51dde4e06d1c492c'
             '560c296946217372ea4357a4eea1a062e9cb8e21acb40fe842259baa18aa8da9'
@@ -89,7 +90,8 @@ sha256sums=('e342b04a2aa63808ea0ef1baab28fc520bd031ef8cf93d9ee4a31d4058fcb622'
             '035ea4b2a7621054f4560471f45336b981538a40172d8f17285910d4e0e0b3ef'
             '5d55bb51c3cade17dde2fec87a385604780afd117dbd01dfbb4b1b83f1ae2660'
             '2d9260b80b43bbd605cf420d6bd53aa7262103dfd77196ba590ece5600b6dc0d'
-            '2a47402a0fd9a0f9e6c9e83f25370e781b09c390d1d724320b756d7880c80416')
+            '2a47402a0fd9a0f9e6c9e83f25370e781b09c390d1d724320b756d7880c80416'
+            'a98c04e6a215a99bbb6bce3f572245235edcb0497d46fb33beea102fa4f25e36')
 prepare() {
   cd "${srcdir}/linux-${_basekernel}"
 
@@ -159,6 +161,9 @@ prepare() {
   # set extraversion to pkgrel
   sed -ri "s|^(EXTRAVERSION =).*|\1 -${pkgrel}|" Makefile
 
+  # apply custom config fragments
+  KCONFIG_CONFIG="./.config" "[merge_script]" -m "./.config" "${srcdir}/preempt.config"
+
   # don't run depmod on 'make install'. We'll do this ourselves in packaging
   sed -i '2iexit 0' scripts/depmod.sh
 
